proposal/package-importer.d.ts.md

```diff
@@ -0,0 +1,386 @@
+> or extensions, except where specified. When using the `pkg:` url scheme,
+> authors need to explicitly use the fully resolved filename.
+
+## Procedures
+
+### Resolving a `pkg` URL
+
+This algorithm takes a URL with scheme `pkg`, and an optional URL `previousURL`. It returns a canonical file path or null.
+
+- Let `fullPath` be `url`'s path.
+- Let `packageName` be the result of [resolving a package name], and `subPath`
+  be the path without the `packageName`.
+- Let `packageRoot` be the result of [resolving the root directory for a
+  package].
+- Let `packageManifest` be the result of loading the `package.json` at
+  `packageRoot`.
+- If `packageManifest` is not set, throw an error.
+- Let `resolved` be the result of [Resolving package exports] to resolve `fullPath`
+  with the `sass` condition set, using `packageManifest`.
+  - If `resolved` has the scheme `file:` and an extension of `sass`, `scss` or
+    `css`, return it.
+  - Otherwise, if `resolved` is not null, throw an error.
+- Let `resolved` be the result of [Resolving package exports] to resolve `fullPath`
+  with the `style` condition set, using `packageManifest`.
+  - If `resolved` has the scheme `file:` and an extension of `sass`, `scss` or
+    `css`, return it.
+  - Otherwise, if `resolved` is not null, throw an error.
+- If `subPath` is empty, return result of [resolving package root values].
+- Let `resolved` be `subPath` resolved relative to `packageRoot`.
+- Return the result of [resolving a file url] with `resolved`.
+
+
+[previous url]: ../accepted/prev-url.d.ts.md
+[Resolving package exports]: #resolving-package-exports
+[resolving package root values]: #resolving-package-root-values
+[resolving a package name]: #resolving-a-package-name
+[parsing a url]: https://url.spec.whatwg.org/#concept-url-parser
+[resolving the root directory for a package]:
+    #resolving-the-root-directory-for-a-package
+[resolving a file url]: ../spec/modules.md#resolving-a-file-url
+
+### Resolving package root values
+
+This algorithm takes a string `packagePath` which is the root directory for a
+package and `packageManifest`, which is the contents of the `package.json` file,
+and returns a file URL.
+
+- Let `sassValue` be the value of `sass` in `packageManifest`.
+- If `sassValue` is a relative path with an extension of `sass`, `scss` or
+  `css`, return the `packagePath` appended with `sassValue`.
+- Let `styleValue` be the value of `style` in `packageManifest`.
+- If `styleValue` is a relative path with an extension of `css`, return the
+  `packagePath` appended with `styleValue`.
+- Otherwise return the result of [resolving a file url] with `packagePath`.
+
+[resolving the root directory for a package]:
+    #resolving-the-root-directory-for-a-package
+[resolving a file url]: ../spec/modules.md#resolving-a-file-url
+
+### Resolving a package name
+
+This algorithm takes a string, `url`, and returns the portion that identifies
```

Comment:
This only takes the _path_ portion of the URL, right? If so, naming it `url` is a little misleading.